app/controllers/application_controller.rb

```diff
@@ -2,4 +2,39 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :current_user
+
+  def current_order
+    return @order if @order
+    if !session[:order_id].nil?
+      @order = Order.find(session[:order_id])
+    else
+      @order = Order.create
+      @order.update(status: "pending")
+      session[:order_id] = @order.id
+      return @order
+    end
+  end
+
+	def current_user
```

Comment:
This method should be indented on the same level as the rest of the methods in this file. It looks like this bit of code might be using tab characters rather than space characters, possibly making it show up as properly indented in Atom...